applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1481,8 +1487,139 @@ void BaseSolidElement::CalculateConstitutiveVariables(
     mConstitutiveLawVector[PointNumber]->CalculateMaterialResponse(rValues, ThisStressMeasure); //here the calculations are actually done
+
+    // We undo the rotation of strain/F, C, stress
+    RotateToGlobalAxes(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseSolidElement::RotateToLocalAxes(
+    ConstitutiveLaw::Parameters& rValues
+    )
+{
+    if (this->Has(LOCAL_AXIS_1)) {
+        if (this->Has(LOCAL_AXIS_2)) {
+            if (this->Has(LOCAL_AXIS_3)) {
+                const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+                BoundedMatrix<double, 3, 3> rotation_matrix;
+
+                const BoundedVector<double, 3> local_axis_1 = this->GetValue(LOCAL_AXIS_1);
+                const BoundedVector<double, 3> local_axis_2 = this->GetValue(LOCAL_AXIS_2);
+                const BoundedVector<double, 3> local_axis_3 = this->GetValue(LOCAL_AXIS_3);
+                if (MathUtils<double>::Norm3(local_axis_1) > 1.0001 ||
+                    MathUtils<double>::Norm3(local_axis_2) > 1.0001 ||
+                    MathUtils<double>::Norm3(local_axis_2) > 1.0001) {
+                        KRATOS_ERROR << "The norm of one of the LOCAL_AXIS is greater than 1.0!" << std::endl;
+                    }
+
+                rotation_matrix(0, 0) = local_axis_1(0); rotation_matrix(0, 1) = local_axis_2(0); rotation_matrix(0, 2) = local_axis_3(0);
+                rotation_matrix(1, 0) = local_axis_1(1); rotation_matrix(1, 1) = local_axis_2(1); rotation_matrix(1, 2) = local_axis_3(1);
+                rotation_matrix(2, 0) = local_axis_1(2); rotation_matrix(2, 1) = local_axis_2(2); rotation_matrix(2, 2) = local_axis_3(2);
+
+                if (UseElementProvidedStrain()) { // we rotate strain
+                    if (strain_size == 6) {
+                        BoundedMatrix<double, 6, 6> voigt_rotation_matrix;
+                        ConstitutiveLawUtilities<6>::CalculateRotationOperatorVoigt(rotation_matrix, voigt_rotation_matrix);
+                        rValues.GetStrainVector() = prod(voigt_rotation_matrix, rValues.GetStrainVector());
+                    } else if (strain_size == 3) {
+                        BoundedMatrix<double, 3, 3> voigt_rotation_matrix;
+                        ConstitutiveLawUtilities<3>::CalculateRotationOperatorVoigt(rotation_matrix, voigt_rotation_matrix);
+                        rValues.GetStrainVector() = prod(voigt_rotation_matrix, rValues.GetStrainVector());
+                    }
+                } else { // rotate F
+                    BoundedMatrix<double, 3, 3> inv_rotation_matrix;
+                    double aux_det;
+                    MathUtils<double>::InvertMatrix3(rotation_matrix, inv_rotation_matrix, aux_det);
+                    const auto &r_F = rValues.GetDeformationGradientF();
```

Comment:
```suggestion
                    const auto& r_F = rValues.GetDeformationGradientF();
```
I prefer the other way...